libnmstate/nm/ovs.py

```diff
@@ -170,6 +176,10 @@ def _get_bridge_port_info(port_profile, devices_info):
 def _get_bridge_options(bridge_device):
     bridge_options = {}
     con = connection.get_device_connection(bridge_device)
+    if not con:
+        logging.debug('No active connection detected for dev %s',
+                      bridge_device)
+        con = bridge_device.get_available_connections()[0]
```

Comment:
What is the idea here?

Reply:
This is indeed strange, but for an ovs bridge device, it may exist as a profile without a backing active connection.
It is enough to pass the tests now, but I expect this to need some tweaking to pass some other edge scenarios.

I hope to be guided by the tests/scenarios, anything else is just guessing. Intuitive it is not.

Reply:
This should somehow be documented then.

Reply:
IMHO the logic which connection to select (in this case the first available connection) should be part of the `nm.applier` code instead of the different device modules to ensure that they all take the same choice.

Reply:
This is an interesting issue.
- Currently, the applier is dependent and calling the individual modules. The specific modules cannot depend on it as it will cause a dependency loop. We could reverse the dependency as a refactoring step, making the individual modules a plugin to the nm applier.
- I'm not convinced there is a shared logic regarding which connection to use at the moment. In most cases, when a profile is present and relevant, it is activated and appears in 'activated' state. With OVS it is not following that behaviour, so an adjustment was needed.
- Regarding documentation, I was hoping that the debug message is enough. It states why we are in this condition and then the `bridge_device.get_available_connections()[0]` is self explanatory. I guess I can wrap this in a special function and document it, but I am not sure how to name it or what to exactly say.
We could also comment this in the commit message, will that satisfy your request here?

Reply:
Not sure, why there needs to be a dependency loop.My idea is that `get_ovs_info()` is called with a profile as its argument instead of the device name. But now that I think of this, `_get_bridge_options()` should not access a non-active profile to get the options since currently Nmstate only reports the runtime state and not the configuration state. The information from `bridge_device.get_available_connections()[0]` is not guaranteed to match the actual runtime state. I guess the current code is a result of the problem that NM sometimes forgets that it manages a OVS bridge. Therefore the device is there but there is no active connection.

Reply:
About the docs: The justification, why accessing the first available profile is correct, is missing.

Reply:
I do not understand how that is related to the 'applier' then, the `get_ovs_info` is called from netinfo, not from the applier.
Not accessing a non active profile is a correct general note, but NM is not creating a bridge if no interface is connected to it. I can reflect that using an `xfail` test instead. Is this what you prefer?

Doc related: The justification or explanation is in the commit message, telling long stories in the code is not a recommended practice (it rots).